### DISCUSSION TITLE
Support for windows

### DISCUSSION THREAD

**Discussion by amityahav** [asciinema/discussions#244] [General]
Hey @ku1ik does recording is supported for windows terminals?

Thanks

**Comment by ku1ik** [comment]
There's no official support at the moment but I heard it works fine under WSL. Neither me nor other occasional contributors use Windows so this was never something we could reliably maintain.

To get proper Windows supports we'd need to make the recording part of the code use some unix-compatible API, probably ConPTY.

I haven't used Windows systems since about 2003 and don't have one at hand to play with it, but if someone wants to prototype ConPTY-based recording in the new Rust codebase (https://github.com/asciinema/asciinema/tree/develop/src) then we can consider including it in 3.0 or later.

**Reply by amityahav** [reply]
@ku1ik thanks for the reply! Actually I was talking about the avt component. Will it emulate properly windows ascii codes, etc.. ? Not really familiar with this is there any different from Linux shells?

**Reply by ku1ik** [reply]
I'm not quite sure, but from what I've read I understand that ConPTY emits xterm-compatible control sequences, which means it should just work if we used ConPTY to capture the output stream and then fed it to avt. I haven't tested this though.
